### PR TITLE
feat(NodeAllocator): Allocate containers

### PR DESCRIPTION
**Pull request by jmcadden**
Adds a DockerDescriptor class, internal to it is the logic to start and
clean-up the container.

AllocateContainer starts a container and adds it to its network

### CODE REVIEW

**Comment by jmcadden** [comment]
This is good to go

File: hosted/src/NodeAllocator.cc [hunk 1]
```diff
@@ -28,6 +29,53 @@ int ebbrt::NodeAllocator::DefaultNumaNodes;
...
+  std::cerr << "exec: " << cmd << std::endl;
+#endif
+  ebbrt::kbugon(cmd.size() == 0);
+  std::string out;
+  char line[kLineSize];
+  auto f = popen(cmd.c_str(), "r");
+  if (f == nullptr) {
+    throw std::runtime_error("Failed to run command: " + cmd);
+  }
+  while (std::fgets(line, kLineSize, f)) {
```
Comment (handong32) [inline hosted/src/NodeAllocator.cc]: shouldn' it be fgets(...) != NULL

File: hosted/src/NodeAllocator.cc [hunk 2]
```diff
@@ -28,6 +29,53 @@ int ebbrt::NodeAllocator::DefaultNumaNodes;
...
+    std::string rm = std::string("docker rm -f ") + cid_;
+    RunCmd(stop);
+    RunCmd(rm);
+  }
+}
+
+std::string ebbrt::NodeAllocator::DockerContainer::Start() {
+  if (!cid_.empty() || img_.size() == 0) {
+    throw std::runtime_error("Error: attempt to start unspecified container ");
+    return std::string();
```
Comment (handong32) [inline hosted/src/NodeAllocator.cc]: can't return some error string?

**Comment by jmcadden** [comment]
@BetsyRT r+

**Comment by BetsyRT** [comment]
:pushpin: Commit 5295e9c has been approved by `jmcadden`

**Comment by BetsyRT** [comment]
:hourglass: Testing commit 5295e9c with merge 42c7313...

**Comment by BetsyRT** [comment]
:sunny: Test successful - [runtests](http://192.168.73.2:8010/builders/runtests/builds/49)
Approved by=jmcadden
Pushing 42c73138d2c12408d47737818db07e1b810fb753 to master...

**Pull request merged**